crypto/Kconfig

```diff
@@ -751,6 +751,14 @@ config CRYPTO_GHASH
 	  GHASH is the hash function used in GCM (Galois/Counter Mode).
 	  It is not a general-purpose cryptographic hash function.
 
+config CRYPTO_POLYVAL
+	tristate "POLYVAL hash function"
```

Comment:
I don't think there's much reason to make this option user-selectable, as opposed to just making it be selected by `CRYPTO_HCTR2`, and by the architecture-specific POLYVAL implementations if they'll rely on this as a fallback.  (Having a prompt string after `tristate` makes it user-selectable; having just `tristate` makes it non-user-selectable.)  See what's done for `CRYPTO_NHPOLY1305`, for example.